File: tests/fixtures/simple.py
#!/usr/bin/env python3
"""Simple test program for debugger integration tests"""

import sys

def add(a: int, b: int) -> int:
    # BREAKPOINT_MARKER: add_body
    result = a + b
    return result

def factorial(n: int) -> int:
    # BREAKPOINT_MARKER: factorial_body
    if n <= 1:
        return 1
    return n * factorial(n - 1)

def main():
    # BREAKPOINT_MARKER: main_start
    x = 10
    y = 20

    # BREAKPOINT_MARKER: before_add
    sum_result = add(x, y)
    print(f"Sum: {sum_result}")

    # BREAKPOINT_MARKER: before_factorial
    fact = factorial(5)
    print(f"Factorial: {fact}")

    # BREAKPOINT_MARKER: before_exit
    return 0

if __name__ == "__main__":
    # BREAKPOINT_MARKER: entry_point
    sys.exit(main())

File: tests/fixtures/simple.rs
// Simple test program for debugger integration tests

fn add(a: i32, b: i32) -> i32 {
    // BREAKPOINT_MARKER: add_body
    let result = a + b;
    result
}

fn factorial(n: i32) -> i32 {
    // BREAKPOINT_MARKER: factorial_body
    if n <= 1 {
        return 1;
    }
    n * factorial(n - 1)
}

fn main() {
    // BREAKPOINT_MARKER: main_start
    let x = 10;
    let y = 20;

    // BREAKPOINT_MARKER: before_add
    let sum = add(x, y);
    println!("Sum: {}", sum);

    // BREAKPOINT_MARKER: before_factorial
    let fact = factorial(5);
    println!("Factorial: {}", fact);

    // BREAKPOINT_MARKER: before_exit
}
